src/test/kotlin/app/services/impl/HttpKeyServiceTest.kt

```diff
@@ -134,48 +156,71 @@ class HttpKeyServiceTest {
         Assert.assertEquals("PLAINTEXT_DATAKEY", dataKeyResult)
         keyService.decryptKey("123", "ENCRYPTED_KEY_ID")
         verify(httpClient, times(1))
-            .execute(ArgumentMatchers.any(HttpPost::class.java))
+            .execute(any(HttpPost::class.java))
     }
 
     @Test(expected = DataKeyDecryptionException::class)
-    fun testDecryptKeyBadKey() {
+    fun testDecryptKey_WithABadKey_WillNotRetry() {
         val statusLine = mock(StatusLine::class.java)
         given(statusLine.statusCode).willReturn(400)
         val httpResponse = mock(CloseableHttpResponse::class.java)
         given(httpResponse.statusLine).willReturn(statusLine)
         val httpClient = mock(CloseableHttpClient::class.java)
-        given(httpClient.execute(ArgumentMatchers.any(HttpPost::class.java))).willReturn(httpResponse)
+        given(httpClient.execute(any(HttpPost::class.java))).willReturn(httpResponse)
         given(httpClientProvider.client()).willReturn(httpClient)
+
+        keyService.decryptKey("123", "ENCRYPTED_KEY_ID")
+
+        verify(httpClient, times(1))
+            .execute(any(HttpPost::class.java))
+    }
+
+    @Test(expected = DataKeyServiceUnavailableException::class)
+    fun testDecryptKey_ServerError_WillCauseRetry() {
+
+        val statusLine = mock(StatusLine::class.java)
+        given(statusLine.statusCode).willReturn(503)
+        val httpResponse = mock(CloseableHttpResponse::class.java)
+        given(httpResponse.statusLine).willReturn(statusLine)
+        val httpClient = mock(CloseableHttpClient::class.java)
+        given(httpClient.execute(any(HttpPost::class.java))).willReturn(httpResponse)
+        given(httpClientProvider.client()).willReturn(httpClient)
+
         keyService.decryptKey("123", "ENCRYPTED_KEY_ID")
+
+        verify(httpClient, times(10))
+            .execute(any(HttpPost::class.java))
     }
 
     @Test(expected = DataKeyServiceUnavailableException::class)
-    fun testDecryptKeyServerError() {
+    fun testDecryptKey_UnknownHttpError_WillCauseRetry() {
 
         val statusLine = mock(StatusLine::class.java)
         given(statusLine.statusCode).willReturn(503)
         val httpResponse = mock(CloseableHttpResponse::class.java)
         given(httpResponse.statusLine).willReturn(statusLine)
         val httpClient = mock(CloseableHttpClient::class.java)
-        given(httpClient.execute(ArgumentMatchers.any(HttpPost::class.java))).willReturn(httpResponse)
+        given(httpClient.execute(any(HttpPost::class.java))).willThrow(RuntimeException("Boom!"))
         given(httpClientProvider.client()).willReturn(httpClient)
+
         keyService.decryptKey("123", "ENCRYPTED_KEY_ID")
+
+        verify(httpClient, times(10))
+            .execute(any(HttpPost::class.java))
     }
 
     @Autowired
     private lateinit var keyService: HttpKeyService
 
     @Autowired
     private lateinit var httpClientProvider: HttpClientProvider
```

Comment:
Looks like more than one line end of the file

Reply:
Done, thanks